docs/_posts/2022-06-24-types-as-comments-proposal.md

```diff
@@ -0,0 +1,170 @@
+
+Part of the appeal of TypeScript is that it adds this additional layer of static—i.e. before runtime—safety to your JavaScript code.
+
+> "The main benefit of TypeScript is that it can highlight unexpected behavior in your code, lowering the chance of bugs."
+> 
+> — [TypeScript in 5 Minutes](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes.html)
+
+Let's now, for example, directly run the above code examples with TypeScript (using `ts-node`).
+
+```bash
+$ ts-node -e 'console.log(3 + [])'
+[eval].ts:1:13 - error TS2365: Operator '+' cannot be applied to types 'number' and 'never[]'.
+
+1 console.log(3 + [])
+              ~~~~~~
+$ ts-node -e 'console.log({}.foo.bar)'
+[eval].ts:1:16 - error TS2339: Property 'foo' does not exist on type '{}'.
+
+1 console.log({}.foo.bar)
+                 ~~~
+```
+
+As we can see, this type of error can easily be detected before running the code by applying type checking. In the first case, TypeScript tells us that we should reconsider adding a number and an array. In the second case, we are informed that empty objects do not magically come with an arbitrary property called `foo`. Both are very reasonable properties of safe JavaScript code and it's extremely helpful that TypeScript checks them for us before we run the code.
+
+Unsurprisingly, TypeScript is considered one of the technologies that are considered safe to adopt by the JavaScript developer community, i.e. have both a  high satisfaction and usage level (cf. [State of JS: Satisfaction vs Usage](https://2020.stateofjs.com/en-US/technologies/#scatterplot_overview)).
+
+![State of JS 2020: Satisfaction vs Usage](/assets/2022-06_state-of-js-satisfaction-vs-usage.png)
+
+TODO note on type inference makes explicit static typing optional
+
+## Making type annotations part of JavaScript proper
+
+### Motivation
+
+Having established that static typing is generally useful for JavaScript development (and development in general), the question to address is this: why not just develop in TypeScript if we want to add static type checks to our JavaScript development process? 
+
+One big argument in favor of types as comments in JavaScript compared to TypeScript (or similar languages and tools) is that it gives us the choice of type checker we want to use. We could for example opt for [Flow](https://flow.org/) type checker instead of TypeScript's type checker if we preferred that (and if Flow started to support the types as comments syntax).
+
+Another important argument pro types as comments in JavaScript is that it lets us skip the inevitable JavaScript transpilation step. Compare that to developing in TypeScript, where the usual workflow includes running our code through `tsc` (the TypeScript compiler) before executing it with `node` or in the browser. Dropping the transpilation step may significantly improve the speed of our development loop and moreover may require less set-up (such as the `tsconfig.json` file).
```

Comment:
I'd argue that the one time setup for ts is neglectable.
If you have a tsconfig, you're probably using more than just static typing.

Also: What about further configuration and linting? Typescript e.g. allows to enforce typing (e.g. strict , no implicit any, ...). A typing system that is completely optional is not a good type system.

Reply:
I think you are somewhat missing the point. (Which ofc may be because I'm not clear enough.) There are two arguments here for having type information as (optional) part of the regular JavaScript syntax: 1. it gives you choice of type checker (implicitely, you can pick the rules of the type system that should be applied to your code); 2. the transpilation step can be removed from you development loop.

> A typing system that is completely optional is not a good type system.

I don't agree. Even though the type information may not be checked (i.e. not analysed within the context of a given type system), you still add some information about the intention of your code that is relevant to the human reader.

Reply:
> you still add some information about the intention of your code that is relevant to the human reader.

But isn't this all about static type checking? For adding information for the human reader jsdoc should already be sufficient.

Reply:
I don’t think the proposal is about type checking per se. They specifically say that a default type checker for JavaScript isn’t the goal. It‘s more about having type hints as in Python which may or not be type checked via tooling.